Applications/Utils/ModelPreparation/PartitionMesh/MeshPartitioning.cpp

```diff
@@ -0,0 +1,493 @@
+        {
+            nodes_local_ids_partition.push_back(0);
+        }
+
+        for (std::size_t i = partition_start_node_id[ipart];
+                i < end_non_ghost_node_id[ipart]; i++)
+        {
+            const unsigned node_id = partition_nodes[i]->getID();
+            node_global_ids[node_id] = global_node_id_offset;
+            nodes_local_ids_partition[node_id] = local_id;
+            local_id++;
+            global_node_id_offset++;
+        }
+        // Assign local IDs to ghost nodes
+        for (std::size_t i = end_non_ghost_node_id[ipart];
+                i < partition_end_node_id[ipart]; i++)
+        {
+            nodes_local_ids_partition[partition_nodes[i]->getID()] = local_id;
+            local_id++;
+        }
+
+        // Count the number of integer variables used to define non-ghost elements
+        const PetscInt num_non_ghost_elems = partition_regular_elements.size();
+        PetscInt nmb_element_idxs = 3 * num_non_ghost_elems;
+        for (PetscInt j=0; j<num_non_ghost_elems; j++)
+        {
+            nmb_element_idxs += partition_regular_elements[j]->getNNodes();
+        }
+        const PetscInt num_ghost_elems = partition_ghost_elements.size();
+        PetscInt nmb_element_idxs_g = 3 * num_ghost_elems;
+        for (PetscInt j=0; j<num_ghost_elems; j++)
+        {
+            nmb_element_idxs_g += partition_ghost_elements[j]->getNNodes();
+        }
+
+        std::string ipart_str = std::to_string(ipart);
+
+        // Number of active elements
+        const PetscInt offset_e = num_non_ghost_elems + nmb_element_idxs;
+        const PetscInt offset_e_g = num_ghost_elems + nmb_element_idxs_g;
+
+        // Reserved for nodes for high order interpolation
+        const PetscInt num_non_ghost_nodes_extra = 0;
+
+        const PetscInt num_nodes_linear_element =   partition_end_node_id[ipart]
+                - partition_start_node_id[ipart];
+        const PetscInt num_nodes_quadratic_element = 0;
+
+        const PetscInt num_nodes_global_mesh = nnodes;
+        // Reserved for nodes for high order interpolation
+        const PetscInt num_nodes_global_mesh_extra = nnodes;
+
+        // Write configuration data and elements of this partition
+        head[0] = num_nodes_linear_element + num_nodes_quadratic_element;
+        head[1] = num_nodes_linear_element;
+        head[2] = num_non_ghost_elems;
+        head[3] = num_ghost_elems;
+        head[4] = num_non_ghost_nodes;
+        head[5] = num_non_ghost_nodes + num_non_ghost_nodes_extra; // active nodes
+        head[6] = num_nodes_global_mesh;
+        head[7] = num_nodes_global_mesh_extra;
+        head[8] = nmb_element_idxs;
+        head[9] = nmb_element_idxs_g;
+        if (output_binary)
+        {
+            fwrite(head, 1, num_headers * sizeof(PetscInt), of_bin_cfg);
+
+            head[10] += head[0] * sizeof(NodeStruct);
+            head[11] += offset_e * sizeof(PetscInt);
+            head[12] += offset_e_g * sizeof(PetscInt);
```

Comment:
Yes, it records the starting positions in three big vectors, e.g.. ele_info, which hold all integer variables that define all elements of each partition. I will add a details doc. later on.